Add Unizen staking contract

minapham17 opened the pull request:
Add Unizen staking contract on Polygon network

alokt commented:
@minapham17 can you please resolve conflicts?

minapham17 commented:
@alokt conflict resolved. thank you

alokt commented:
@minapham17 is it working locally? because test case is failing. can you please check
https://github.com/abridged/collabland-staking-contracts/actions/runs/9755711957/job/26946306312?pr=243

minapham17 commented:
@alokt I fixed the issue, and now the test should work. During the testing process, I noticed something important:

For the same static call to our contract, it seems that the RPC is not very stable during the test. This means it can succeed at one moment but fail a few seconds later if I run the test again.

To handle this, I added a try-catch block to prevent failed tests due to the RPC status. Thank you for your review and please let me know if I need to change anything.

minapham17 commented:
@alokt just reminder in case you missed my comment
